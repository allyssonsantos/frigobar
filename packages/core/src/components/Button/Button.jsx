import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Icon from '@material-ui/core/Icon';

const StyledIcon = styled(Icon)`
  margin-right: ${({ size }) => (size === 'large' ? '8px' : '6px')};
`;

const Btn = styled.button`
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-size: 0.875rem;
  transition: all 200ms ease-in-out;

  ${({
    theme: {
      components: {
        button: { backgroundColor, textColor, sizes },
      },
    },
    skin,
    icon,
    size,
    disabled,
    full,
  }) => `
    background-color: ${backgroundColor[skin]};
    color: ${textColor.enabled};
    font-size: ${sizes[size].font}rem;
    padding: ${sizes[size].padding.top}px ${sizes[size].padding.right}px ${
    sizes[size].padding.bottom
  }px ${sizes[size].padding.left}px;

    ${
      disabled
        ? `
      background-color: ${backgroundColor.disabled};
      color: ${textColor.disabled};
      cursor: not-allowed;
    `
        : `
        &:hover {
          box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.4);
        }
        `
    }

    ${StyledIcon} {
      font-size: ${sizes[size].icon}rem;
    }

    ${
      icon
        ? `
      display: flex;
      align-items: center;
      justify-content: center;
    `
        : ''
    }

    ${full ? 'width: 100%;' : ''}
  `}
`;

const Button = ({ children, icon, theme, size, full, ...props }) => (
  <Btn icon={icon} theme={theme} size={size} full={full} {...props}>
    {icon && <StyledIcon size={size}>{icon}</StyledIcon>}
    {children}
  </Btn>
);

Button.propTypes = {
  children: PropTypes.node.isRequired,
  skin: PropTypes.oneOf(['primary', 'info', 'success', 'danger', 'warning', 'neutral']),
  disabled: PropTypes.bool,
  icon: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  full: PropTypes.bool,
};

Button.defaultProps = {
  skin: 'neutral',
  disabled: false,
  icon: undefined,
  size: 'small',
  full: false,
};

export default Button;
