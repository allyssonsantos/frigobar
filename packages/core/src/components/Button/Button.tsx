import React from 'react';
import styled from 'styled-components';

interface ButtonProps {
  children: React.ReactNode;
  /** Change background-color */
  skin?: 'primary' | 'info' | 'success' | 'danger' | 'warning' | 'neutral';
  /** change the width, height and font-size values */
  size?: 'small' | 'medium' | 'large' | undefined;
  /** Change disabled state of the button */
  disabled?: boolean;
  /** 100% width */
  full?: boolean;
  /** add border-radius circle */
  rounded?: boolean;
  /** large button */
  large?: boolean;
}

const Button = styled.button<ButtonProps>`
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-size: 0.875rem;
  transition: all 200ms ease-in-out;
  text-decoration: none;

  ${({
    theme: {
      radius,
      components: {
        button: { backgroundColor, textColor, sizes },
      },
    },
    skin,
    size,
    disabled,
    full,
    rounded,
    large,
  }) => `
    background-color: ${backgroundColor[skin]};
    color: ${textColor.enabled};
    font-size: ${large ? '1.125' : sizes[size].font}rem;
    padding: ${
      large
        ? `${sizes[size].padding.top * 2}px
           ${sizes[size].padding.right * 2}px
           ${sizes[size].padding.bottom * 2}px
           ${sizes[size].padding.left * 2}px
        `
        : `${sizes[size].padding.top}px
           ${sizes[size].padding.right}px
           ${sizes[size].padding.bottom}px
           ${sizes[size].padding.left}px
        `
    };

    ${
      disabled
        ? `
      background-color: ${backgroundColor.disabled};
      color: ${textColor.disabled};
      cursor: not-allowed;
    `
        : `
        &:hover {
          box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.4);
        }
        `
    }

    ${full ? 'width: 100%;' : ''}
    ${rounded ? `border-radius: ${radius[3]}px` : ''}
  `}
`;

Button.defaultProps = {
  skin: 'neutral',
  disabled: false,
  size: 'small',
  full: undefined,
  rounded: undefined,
  large: undefined,
};

export default Button;
