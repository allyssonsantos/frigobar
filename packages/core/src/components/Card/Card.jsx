import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  Content,
  Footer,
  Header,
  Media,
  Avatar,
  Title,
  Subtitle,
} from './sub-components';
import defaultTheme from '../../theme';

const Card = styled.section(
  ({
    maxWidth,
    theme: {
      colors,
      components: { card },
    },
  }) => `
  display: inline-block;
  max-width: ${maxWidth};
  width: 100%;

  border-radius: ${card.border.radius}px;
  
  background-color: ${colors.white};
  box-shadow: 0px 2px 5px 0px ${colors.neutral[200]};
`,
);

Card.Content = Content;
Card.Footer = Footer;
Card.Header = Header;
Card.Media = Media;
Card.Avatar = Avatar;
Card.Title = Title;
Card.Subtitle = Subtitle;

Card.displayName = 'Card';

Card.propTypes = {
  maxWidth: PropTypes.string,
};

Card.defaultProps = {
  maxWidth: '400px',
  theme: defaultTheme,
};

export default Card;
