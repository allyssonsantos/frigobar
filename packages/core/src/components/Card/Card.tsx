import styled, { DefaultTheme, StyledComponentBase } from 'styled-components';
import {
  Content,
  Footer,
  Header,
  Media,
  Avatar,
  Title,
  Subtitle,
} from './sub-components';

interface CardProps {
  /** sets max-width of the card wrapper */
  maxWidth: string;
}

interface CardOverload extends StyledComponentBase<'section', DefaultTheme> {
  Content: typeof Content;
  Footer: typeof Footer;
  Header: typeof Header;
  Media: typeof Media;
  Avatar: typeof Avatar;
  Title: typeof Title;
  Subtitle: typeof Subtitle;
}

const Card: CardOverload = Object.assign(
  styled.section<CardProps>(
    ({
      maxWidth = '400px',
      theme: {
        colors,
        components: { card },
      },
    }) => `
  display: inline-block;
  max-width: ${maxWidth};
  width: 100%;

  border-radius: ${card.border.radius}px;
  
  background-color: ${colors.white};
  box-shadow: 0px 2px 5px 0px ${colors.neutral[200]};
`,
  ),
  { Content, Footer, Header, Media, Avatar, Title, Subtitle },
);

Card.displayName = 'Card';

export default Card;
