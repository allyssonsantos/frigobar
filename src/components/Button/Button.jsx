import styled from 'styled-components';
import defaultTheme from '../../theme';

const Button = styled.button`
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-size: 14px;
  padding: 6px 14px;
  transition: all 200ms ease-in-out;

  &:hover {
    box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.4);
  }

  ${({ theme, skin }) => `
    background-color: ${theme.button[skin].bgColor}
    color: ${theme.button[skin].textColor}
  `}
`;

Button.defaultProps = {
  theme: defaultTheme,
  skin: 'neutral',
};

export default Button;
