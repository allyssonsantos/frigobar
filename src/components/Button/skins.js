const getButtonColors = (colors, skin) => ({
  bgColor: colors[skin][500],
  textColor: colors.neutral[50],
});

const skins = ({ colors }) => ({
  primary: getButtonColors(colors, 'primary'),
  info: getButtonColors(colors, 'info'),
  success: getButtonColors(colors, 'success'),
  warning: getButtonColors(colors, 'warning'),
  danger: getButtonColors(colors, 'danger'),
  neutral: getButtonColors(colors, 'neutral'),
});

export default skins;
