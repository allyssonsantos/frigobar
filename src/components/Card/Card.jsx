import React from 'react';
import styled from 'styled-components';
import { defaultTheme as theme } from '../../theme'

const Card = styled.section`
  border-radius: 4px;
  padding: 16px;

  ${({ theme: { colors } }) => `
    border: 1px solid ${colors.neutral[200]};
  `}
`;

Card.defaultProps = {
  theme,
};

export default Card;
