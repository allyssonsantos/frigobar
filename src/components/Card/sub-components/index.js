import Header from './Header';
import Content from './Content';
import Footer from './Footer';
import Media from './Media';
import Avatar from './Avatar';
import Title from './Title';
import Subtitle from './Subtitle';

export { Header, Content, Footer, Media, Avatar, Title, Subtitle };
