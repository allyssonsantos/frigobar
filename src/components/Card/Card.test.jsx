import React from 'react';
import { render } from '@testing-library/react';

import Card from './Card';
import Button from '../Button';

describe('<Card />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <Card maxWidth="400px">
          <Card.Header avatar="https://dummyimage.com/72x72">
            <Card.Title>Title</Card.Title>
            <Card.Subtitle>Subtitle</Card.Subtitle>
          </Card.Header>
          <Card.Media src="https://dummyimage.com/600x600" />
          <Card.Content>Lorem</Card.Content>
          <Card.Footer>
            <Button skin="primary" style={{ marginRight: 10 }}>
              Button
            </Button>
            <Button skin="primary">Cancel</Button>
          </Card.Footer>
        </Card>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
