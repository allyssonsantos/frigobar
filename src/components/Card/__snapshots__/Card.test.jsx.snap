// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Snapshots should match snapshot 1`] = `
.c2 {
  border-radius: 4px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px 16px 8px;
}

.c3 {
  padding-left: 12px;
}

.c8 {
  padding: 8px 16px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px 16px 16px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c6 {
  margin-bottom: 8px;
  margin-top: 8px;
  overflow: hidden;
  padding-bottom: 56.25%;
  position: relative;
}

.c7 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 100%;
}

.c4 {
  font-size: 1.5rem;
  margin: 0;
}

.c5 {
  font-size: 1rem;
  margin: 0;
}

.c0 {
  border-radius: 4px;
  display: inline-block;
  max-width: 400px;
  box-shadow: 0px 2px 5px 0px #ccc;
  width: 100%;
}

.c10 {
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-size: 0.875rem;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  background-color: #598aed;
  color: #fdfdfd;
  font-size: 0.875rem;
  padding: 6px 14px;
}

.c10:hover {
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.4);
}

.c10 .c11 {
  font-size: 1rem;
}

<div>
  <section
    class="c0"
  >
    <header
      class="c1"
    >
      <img
        class="c2"
        src="https://dummyimage.com/72x72"
      />
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Title
        </h2>
        <h3
          class="c5"
        >
          Subtitle
        </h3>
      </div>
    </header>
    <div
      class="c6"
    >
      <img
        class="c7"
        src="https://dummyimage.com/600x600"
      />
    </div>
    <div
      class="c8"
    >
      Lorem
    </div>
    <footer
      class="c9"
    >
      <button
        class="c10"
        style="margin-right: 10px;"
      >
        Button
      </button>
      <button
        class="c10"
      >
        Cancel
      </button>
    </footer>
  </section>
</div>
`;

exports[`<Card /> Snapshots should match snapshot with rounded avatar 1`] = `
.c2 {
  border-radius: 50%;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px 16px 8px;
}

.c3 {
  padding-left: 12px;
}

.c8 {
  padding: 8px 16px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px 16px 16px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c6 {
  margin-bottom: 8px;
  margin-top: 8px;
  overflow: hidden;
  padding-bottom: 56.25%;
  position: relative;
}

.c7 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 100%;
}

.c4 {
  font-size: 1.5rem;
  margin: 0;
}

.c5 {
  font-size: 1rem;
  margin: 0;
}

.c0 {
  border-radius: 4px;
  display: inline-block;
  max-width: 400px;
  box-shadow: 0px 2px 5px 0px #ccc;
  width: 100%;
}

.c10 {
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-size: 0.875rem;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  background-color: #598aed;
  color: #fdfdfd;
  font-size: 0.875rem;
  padding: 6px 14px;
}

.c10:hover {
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.4);
}

.c10 .c11 {
  font-size: 1rem;
}

<div>
  <section
    class="c0"
  >
    <header
      class="c1"
    >
      <img
        class="c2"
        src="https://dummyimage.com/72x72"
      />
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Title
        </h2>
        <h3
          class="c5"
        >
          Subtitle
        </h3>
      </div>
    </header>
    <div
      class="c6"
    >
      <img
        class="c7"
        src="https://dummyimage.com/600x600"
      />
    </div>
    <div
      class="c8"
    >
      Lorem
    </div>
    <footer
      class="c9"
    >
      <button
        class="c10"
        style="margin-right: 10px;"
      >
        Button
      </button>
      <button
        class="c10"
      >
        Cancel
      </button>
    </footer>
  </section>
</div>
`;
